tests/models/test_anthropic.py

```diff
@@ -41,28 +52,67 @@
     pytest.mark.anyio,
 ]
 
+# Type variable for generic AsyncStream
+T = TypeVar('T')
+
 
 def test_init():
     m = AnthropicModel('claude-3-5-haiku-latest', api_key='foobar')
     assert m.client.api_key == 'foobar'
     assert m.name() == 'anthropic:claude-3-5-haiku-latest'
 
 
+class MockAsyncStream(AsyncStream[T]):
+    """Mock implementation of AsyncStream for testing."""
+
+    def __init__(self, events: list[list[T]]):
```

Comment:
To fulfill the contract of AsyncAnthropic, we need to pass `create_mock` a single instance of our MockAsyncStream. However since this async stream is recycled across a potentially multi-turn conversation with tool calls, we also need the ability to provide multiple chained sequences together.

We modify the existing MockAsyncStream class here to support list[list[Messages]] instead of a single event stream. If desired this change could be refactored into the original class.